numpy/f2py/cfuncs.py

```diff
@@ -481,18 +481,18 @@
 """
 needs['STRINGCOPYN'] = ['string.h', 'FAILNULL']
 cppmacros['STRINGCOPYN'] = """\
-#define STRINGCOPYN(to,from,buf_size)                           \\
+/*
+STRINGCOPYN copies N bytes.
+
+`to` and `from` buffers must have sizes of at least N bytes.
+*/
+#define STRINGCOPYN(to,from,N)                                  \\
     do {                                                        \\
-        int _m = (buf_size);                                    \\
+        int _m = (N);                                           \\
         char *_to = (to);                                       \\
         char *_from = (from);                                   \\
         FAILNULL(_to); FAILNULL(_from);                         \\
         (void)strncpy(_to, _from, sizeof(char)*_m);             \\
```

Comment:
Note that `sizeof(char)` is defined to be 1 (even if a `char` is multiple bytes!), so arguably this could be just
```suggestion
        (void)strncpy(_to, _from, _m);                          \\
```

Reply:
Fixed. Also in other places.